torch/quantization/fx/quantize.py

```diff
@@ -1302,9 +1302,8 @@ def is_output_quantized(node: Node, obj: QuantizeHandler) -> bool:
                 quantized = node_arg_is_quantized(node.args[0])
 
             # the output is unquantized if the node is not a CopyNode
-            # and activation is fp16 (since we will output fp32 currently for fp16
-            # converter
-            if not activation_is_int8_quantized(qconfig) or \
+            # or the activation is not statically quantized
+            if not activation_is_statically_quantized(qconfig) or \
```

Comment:
@vkuzo the main change made in the PR is described in the summary already, but what's causing the change in test case is this line. previously we are saying if qconfig is not int8 we'll set quantized to False (since fp16 QuantizeHandler will output float32 Tensor), but in reality that is not correct. In the reshape_fp16 test case, we have:
```
dequantize = linear.dequantize();  linear = None
to = dequantize.to(torch.float16);  dequantize = None
reshape = to.reshape(-1, 4);  to = None
```
that is, the output of reshape is float16 (quantized to float16) instead of non-quantized.

after this PR the QuantizeHandler for reshape is producing a quantized Tensor as well (same as int8 QuantizeHandler), so we can change this condition to cover all quantized cases.